transport/http/outbound_test.go

```diff
@@ -453,3 +453,78 @@ func TestOutboundNoDeadline(t *testing.T) {
 	_, err := out.call(context.Background(), &transport.Request{})
 	assert.Equal(t, yarpcerrors.Newf(yarpcerrors.CodeInvalidArgument, "missing context deadline"), err)
 }
+
+func TestSvcMatchSuccess(t *testing.T) {
+	matchServer := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, req *http.Request) {
+			defer req.Body.Close()
+			w.Header().Set(ServiceHeader, req.Header.Get(ServiceHeader))
+			_, err := w.Write([]byte("Service name header return"))
+			assert.NoError(t, err)
+		},
+	))
+	defer matchServer.Close()
+
+	httpTransport := NewTransport()
+	out := httpTransport.NewSingleOutbound(matchServer.URL)
+	require.NoError(t, out.Start(), "failed to start outbound")
+	defer out.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), testtime.Second)
+	defer cancel()
+	testSvcName := "Service"
+	_, err := out.Call(ctx, &transport.Request{
+		Service: testSvcName,
+	})
+	require.NoError(t, err)
+}
+
+func TestSvcMatchFailed(t *testing.T) {
+	dismatchServer := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, req *http.Request) {
+			defer req.Body.Close()
+			w.Header().Set(ServiceHeader, "ThisIsAWrongSvcName")
+			_, err := w.Write([]byte("Wrong service name header return"))
+			assert.NoError(t, err)
+		},
+	))
+	defer dismatchServer.Close()
+
+	httpTransport := NewTransport()
+	out := httpTransport.NewSingleOutbound(dismatchServer.URL)
+	require.NoError(t, out.Start(), "failed to start outbound")
+	defer out.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), testtime.Second)
+	defer cancel()
+	testSvcName := "Service"
+	_, err := out.Call(ctx, &transport.Request{
+		Service: testSvcName,
+	})
+	assert.Error(t, err, "expected failure for service name dismatch")
+}
+
+func TestSvcMatchNoHeader(t *testing.T) {
+	noHeaderServer := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, req *http.Request) {
+			defer req.Body.Close()
+			w.Header().Set("rpc-header-ThisIsNotServiceHeader", "hello")
```

Comment:
Instead of setting an arbitrary header, let's just not set anything and write a comment. ie
`// intentionally do not set a response header`

Reply:
Resolved.